docs/components/table-view.md

```diff
@@ -0,0 +1,1016 @@
+- `emptyStateActionMessage` - Button text for empty state action
+- `emptyStateActionRoute` - Route for empty state action
+- `emptyStateButtonAppearance` - Appearance of empty state action button. See [KButton `appearance` prop](/components/button#appearance) for details
+
+:::tip
+Should you want to display an icon inside of action button, you can use `empty-state-action-icon` slot.
+:::
+
+When empty state action button is clicked, KTableView emits the `empty-state-action-click` event.
+
+<KTableView 
+  :data="[]"
+  :headers="basicHeaders()"
+  empty-state-title="No Workspaces exist"
+  empty-state-message="Adding a new Workspace will populate this catalog."
+  empty-state-icon-variant="kong"
+  empty-state-action-message="Create a Workspace"
+  empty-state-action-route="/"
+  empty-state-button-appearance="secondary"
+>
+  <template #empty-state-action-icon>
+    <AddIcon />
+  </template>
+</KTableView>
+
+```html
+<KTableView 
+  :data="[]"
+  :headers="headers"
+  empty-state-title="No Workspaces exist"
+  empty-state-message="Adding a new Workspace will populate this table."
+  empty-state-icon-variant="kong"
+  empty-state-action-message="Create a Workspace"
+  empty-state-action-route="/"
+  empty-state-button-appearance="secondary"
+>
+  <template #empty-state-action-icon>
+    <AddIcon />
+  </template>
+</KTableView>
+```
+
+### Error
+
+Set the `error` prop to `true` to enable the error state.
+
+<KTableView
+  error
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  error
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+Set the following properties to customize error state:
+
+- `errorStateTitle` - Title text for error state
+- `errorStateMessage` - Message for error state
+- `errorStateActionMessage` - Button text for error state action
+- `errorStateActionRoute` - Route for error state action
+
+A `error-action-click` event is fired when error state action button is clicked.
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders()"
+  error
+  error-state-title="Something went wrong"
+  error-state-message="Error loading data."
+  error-state-action-message="Report an Issue"
+  error-state-action-route="/"
+/>
+
+```html
+<KTableView
+  error
+  :data="tableData"
+  :headers="headers"
+  error-state-title="Something went wrong"
+  error-state-message="Error loading data."
+  error-state-action-message="Report an Issue"
+  error-state-action-route="/"
+/>
+```
+
+## Slots
+
+### Column Header
+
+You can slot in your custom content into each column header. For that, use column `key` value prefixed with `column-*` like in the example below.
+
+Slot props:
+* `column` - column header object
+
+:::tip NOTE
+Header slot container is a `display: flex;` element that takes care of spacing between slotted items so you can slot in your items without having to worry about adding margin between them.
+:::
```

Comment:
nit: I'm not sure this is needed; we don't typically call out CSS implementation in the docs; it should be obvious here.
```suggestion
```

Reply:
Agreed